feat : 날짜 포매터 전역 적용

rladmstn opened the pull request:
## 📌 Related Issue
<!-- issue 링크 -->
https://github.com/GAMZA-BAT/algohub-server/issues/120

## 🚀 Description
<!-- 작업 내용 -->
- 기존에 날짜 관련해서 Date 타입으로 반환하거나, `yyyy-MM-dd` 형태이던 필드들을 `yyyy.MM.dd` 로 format 해서 전달하도록 수정했습니다.
<p align="center">
  <img width="189" alt="image" src="https://github.com/user-attachments/assets/e66b811c-7d89-4226-b8c0-dd8798e387c7">
</p>

- 또한 시,분,초 단위의 정보는 필요 없어서 모두 날짜 형태로만 반환하도록 수정했습니다. 
- 전역적으로 사용되는 포매터인만큼, util적 특성을 띈다고 판단해 `DateFormatUtil` 클래스를 생성했습니다.
- `formatDate(LocalDate date)` 메소드가 해당 형태로 변환한 후 String을 반환하도록 만들었습니다.
- 거의 대부분의 Get DTO를 수정했다고 보시면 될 듯 합니다.

- (10/22 수정) 풀이 조회는 `solvedDate`가 `yyyy.MM.dd HH:mm:ss` 형태여야 한다는 요구가 있어, `formatDateTime(LocalDateTime dateTime)` 메소드를 추가 구현했습니다.

## 📢 Review Point

## 📚Etc (선택)
<!-- 작업 도중 생긴 특이사항, 또는 고려할 것들(?) -->

rladmstn committed:
feat : 날짜 관련 response 포매팅 Util 클래스 생성

- 'yyyy-MM-dd' -> 'yyyy.MM.dd'로 수정해 String 반환하는 util 메서드 생성

rladmstn committed:
feat : GetBoardResponse 날짜 포매팅

rladmstn committed:
feat : GetCommentResponse 날짜 포매팅

rladmstn committed:
feat : GetGroupMemberResponse 날짜 포매팅

rladmstn committed:
feat : GetGroupResponse 날짜 포매팅

rladmstn committed:
feat : GetNotificationResponse 날짜 포매팅

rladmstn committed:
feat : GetProblemResponse 날짜 포매팅

rladmstn committed:
feat : GetSolutionResponse 날짜 포매팅

rladmstn committed:
feat : GetStudyGroupResponse 날짜 포매팅

rladmstn committed:
feat : BoardService DTO 생성 시 날짜 포매팅

rladmstn committed:
test : StudyGroupControllerTest DTO에 대해 포매터 적용

rladmstn committed:
test : BoardServiceTest 조회 관련 DTO 날짜 포맷 적용

rladmstn committed:
test : CommentServiceTest 조회 관련 DTO 날짜 포맷 적용

rladmstn committed:
test : ProblemServiceTest 조회 관련 DTO 날짜 포맷 적용

rladmstn committed:
test : SolutionServiceTest 조회 관련 DTO 날짜 포맷 적용

rladmstn committed:
test : StudyGroupServiceTest 조회 관련 DTO 날짜 포맷 적용

hwangjokim reviewed (approved):
db는 더 높은 정밀도 가진 Localate 그대로 사용하고, DTO만 가공하는 것 Best Practice네요 :)

rladmstn committed:
Merge branch 'develop' into feat/#120

rladmstn committed:
feat : 풀이 조회 시 solvedDate 포맷 수정

- `yyyy.MM.dd` => `yyyy.MM.dd HH:mm:ss` 형태로 시간 포함한 포맷으로 수정
- DateformatUtil 클래스에 DateTime을 포매팅하는 함수 formatDateTime 추가

rladmstn committed:
test : solution 조회 시, solvedDate 포매팅 형태 수정

hwangjokim commented:
문득 든 생각인데, DateTime 클래스를 extends하고 toHumanizeString() 같은 메서드를 파서 쓰는게 더 나을까 싶기도 하네요. 더 과격하게 나가면 toString()을 override한다거나..

rladmstn commented:
> 문득 든 생각인데, DateTime 클래스를 extends하고 toHumanizeString() 같은 메서드를 파서 쓰는게 더 나을까 싶기도 하네요. 더 과격하게 나가면 toString()을 override한다거나..

extends 해서 구현하는건 생각 못해봤네요! 

그렇다면 DateTime, Date 각각을 extends 해야 한다는 것 같은데, 비용측면에서도 그렇게 구현하는게 더 나을까요?

저는 뭐가 왜 더 나은지 확실한 설명은 못하겠네요..

sh0723 reviewed (approved):
노가다 엄청 하셨었네요 .. 고생하셨슴다

hwangjokim commented:
좀더 찾아보니까, 유틸이 더 나은 선택인것같아요
그런데 스프링에서는 Jackson (객체를 json과 상호변환 해주는것)에 있는 ObjectMapper를 건드려서, 포매팅을 자동으로 해주는 경우도 많은 것 같아요.
https://perfectacle.github.io/2018/01/16/jackson-local-date-time-serialize/
https://gist.github.com/ksundong/3246e3e01d9ec9b201a49424699eaded
https://interconnection.tistory.com/137

요것도 한번 알아보면 좋을것같네요

rladmstn commented:
> 요것도 한번 알아보면 좋을것같네요
오 저런 어노테이션도 있군요! 기존 포매팅 코드 갖다가 쓰느라 저런 건 찾아볼 생각을 못했네용 정보 감삼다

rladmstn committed:
Resolve conflict

s-hwan commented on src/main/java/com/gamzabat/algohub/feature/studygroup/dto/GetGroupMemberResponse.java:
```diff
@@ -10,17 +11,17 @@
 		String profileImage, Long memberId) {
 		this.nickname = nickname;
-		this.joinDate = joinDate;
-		this.achivement = achivement;
```
ㅋㅋㅋㅋ 스펠링 틀린거 누구야 !

s-hwan reviewed (approved):
굿굿 핳 👍
